app/controllers/application_controller.rb

```diff
@@ -30,4 +30,15 @@ def logged_in?
   def log_out
     reset_session
   end
+
+  def check_login
+    if !logged_in?
+      return authorization_failure
+    end
+  end
+
+  def authorization_failure(msg = "You are not authorized to perform this action.")
```

Comment:
For what it's worth, I think the way you did this is pretty slick.  I've done authorization failure handlers before but never thought of providing an overridable message to the user.